programs/zk-token-proof-tests/tests/process_transaction.rs

```diff
@@ -0,0 +1,548 @@
+        recent_blockhash,
+    );
+    client.process_transaction(transaction).await.unwrap();
+}
+
+async fn test_close_context_state<T: Pod + ZkProofData>(
+    proof_type: ProofType,
+    space: usize,
+    success_proof_data: &T,
+) {
+    let mut context = ProgramTest::default().start_with_context().await;
+    let rent = context.banks_client.get_rent().await.unwrap();
+
+    let client = &mut context.banks_client;
+    let payer = &context.payer;
+    let recent_blockhash = context.last_blockhash;
+
+    let context_state_account = Keypair::new();
+    let context_state_authority = Keypair::new();
+
+    let context_state_info = ContextStateInfo {
+        context_state_account: &context_state_account.pubkey(),
+        context_state_authority: &context_state_authority.pubkey(),
+    };
+
+    let destination_account = Keypair::new();
+
+    // create a proof context state
+    let instructions = vec![
+        system_instruction::create_account(
+            &payer.pubkey(),
+            &context_state_account.pubkey(),
+            rent.minimum_balance(space),
+            space as u64,
+            &zk_token_proof_program::id(),
+        ),
+        verify_proof(proof_type, Some(context_state_info), success_proof_data),
+    ];
+    let transaction = Transaction::new_signed_with_payer(
+        &instructions,
+        Some(&payer.pubkey()),
+        &[payer, &context_state_account],
+        recent_blockhash,
+    );
+    client.process_transaction(transaction).await.unwrap();
+
+    // try to close context state with incorrect authority
+    let incorrect_authority = Keypair::new();
+    let instruction = close_context_state(
+        ContextStateInfo {
+            context_state_account: &context_state_account.pubkey(),
+            context_state_authority: &incorrect_authority.pubkey(),
+        },
+        &destination_account.pubkey(),
+        proof_type,
+    );
+    let transaction = Transaction::new_signed_with_payer(
+        &[instruction],
+        Some(&payer.pubkey()),
+        &[payer, &incorrect_authority],
+        recent_blockhash,
+    );
+    let err = client
+        .process_transaction(transaction)
+        .await
+        .unwrap_err()
+        .unwrap();
+    assert_eq!(
+        err,
+        TransactionError::InstructionError(0, InstructionError::InvalidAccountOwner)
+    );
+
+    // successfully close proof context state
+    let instruction = close_context_state(
+        ContextStateInfo {
+            context_state_account: &context_state_account.pubkey(),
+            context_state_authority: &context_state_authority.pubkey(),
+        },
+        &destination_account.pubkey(),
+        proof_type,
+    );
+    let transaction = Transaction::new_signed_with_payer(
+        &[instruction.clone()],
+        Some(&payer.pubkey()),
+        &[payer, &context_state_authority],
+        recent_blockhash,
+    );
+    client.process_transaction(transaction).await.unwrap();
+}
```

Comment:
Could you add some tests that mix the wrong proof / proof type? e.g. `ProofType::Withdraw` and `PubkeyValidityProofContext`?

And the different ownership mixes discussed elsewhere:
* self-owned proof context account
* closing with self as destination (should fail)
* closing to owner